### DISCUSSION TITLE
Slice to Typescrypt compiler?

### DISCUSSION THREAD

**Discussion by ericsampson** [zeroc-ice/ice#1487] [Ideas]
Hi, is there plans to produce a Slice to Typescript compiler? The enhanced type support vs JS would be awesome, especially given how TS is getting super popular.

Thanks!

**Comment by pepone** [comment]
Hi @ericsampson 

The 3.7 Slice to JavaScript compiler has a `--typescript` option that allows you to generate Typescript definitions from Slice definitions in addition to the JavaScript code.

We introduced this in 3.7.2 release see https://github.com/zeroc-ice/ice/blob/3.7/CHANGELOG-3.7.md#javascript-changes-5

See also the Typescript examples in https://github.com/zeroc-ice/ice-demos/tree/3.7/typescript

**Reply by ericsampson** [reply]
Thanks I missed that!

Maybe it would be worth renaming it to the "Slice JavaScript/Typescript" compiler to help people find it?

Is the functionality described in any of these doc pages? https://doc.zeroc.com/ice/3.6/language-mappings/javascript-mapping


I noticed that this page doesn't mention that CLI flag:

https://doc.zeroc.com/ice/3.6/language-mappings/javascript-mapping/client-side-slice-to-javascript-mapping/slice2js-command-line-options

**Comment by pepone** [comment]
This is new in 3.7, which is why it is not mentioned in the 3.6 pages
